x/dao/keeper/msg_server_distribution_result.go

```diff
@@ -47,11 +53,20 @@ func (k msgServer) DistributionResult(goCtx context.Context, msg *types.MsgDistr
 // Calculate the difference for a set of participants and clear out all past unresolved staged claims.
 func (k msgServer) clearUnresolvedClaims(ctx sdk.Context, start int64) (err error) {
 	// calculate total amounts for current and future claims
-	currentAmounts, err := k.getClaims(ctx, start, ctx.BlockHeight())
+	claims, err := k.getClaims(ctx, start, ctx.BlockHeight())
 	if err != nil {
 		return err
 	}
 
+	currentAmounts := make(map[string]uint64)
+	for address, amount := range claims.challenger {
+		currentAmounts[address] = amount
+	}
+
+	for address, amount := range claims.challengee {
+		currentAmounts[address] = amount
```

Comment:
I am not completely sure, what happens here, but could it be that an address already has a currentAmount it needs to be `+= amount` rather than `= amount`, basically deleting amount collected as a challenger?

Reply:
true, just fixed